soda/bigquery/soda/data_sources/bigquery_data_source.py

```diff
@@ -153,6 +153,12 @@ def sql_get_table_names_with_count(
         where_clause = f"\nWHERE {table_filter_expression} \n" if table_filter_expression else ""
         return f"SELECT table_id, row_count \n" f"FROM {self.dataset_name}.__TABLES__" f"{where_clause}"
 
+    def sql_get_table(self, table_name: str, limit: int = None) -> str:
```

Comment:
I believe if you use a type checker in your IDE (which you should btw even if we don't enforce it on this repo) that the hint should have warned you. `None` cannot be assigned to an `int`. So if we expect `limit` to sometimes be `None` then I would type this as
`int | None` or Optional[None]